tests/apply-phpunit-patches.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+# All commands below must not fail
+set -eu
+
+# Go to root dir
+cd "$(dirname $0)/../"
+
+echo "Running in: $PWD"
+
+find tests/Cases -type f -print0 | xargs -0 sed -i 's/function setUpBeforeClass()/function setUpBeforeClass(): void/g';
+find tests/Cases -type f -print0 | xargs -0 sed -i 's/function setUp()/function setUp(): void/g';
+find tests/Cases -type f -print0 | xargs -0 sed -i 's/function tearDown()/function tearDown(): void/g';
+
+echo "Done"
+
+# Return back to original dir
+cd - > /dev/null
```

Comment:
Looks like a cool work around that deal with the different phpunit versions, but what about tests if we run them locally ?

Reply:
Well, you will have to patch the files first
I know this is not awesome but you need to make some painfull choices when supporting a wide range of php versions

The patches could be applied (making the files non PHP 5 compatible, and only patch files when running tests for php 5)
What do you think about this last idea ?

Reply:
I like the idea, but the execution of the tests should not modify the code...

I see it in two ways:

* Some sort of wrapper that checks the version of php before running the tests, applies a patch if necessary, runs the tests, then restores the patched files.
* Keep the original version of phpunit :) in composer and just apply --ignore-platform-req for php 8 and above.

Reply:
> Keep the original version of phpunit :) in composer and just apply --ignore-platform-req for php 8 and above.

Not possible, old phpunit versions are not php 8 compatible :/

> Some sort of wrapper that checks the version of php before running the tests, applies a patch if necessary, runs the tests, then restores the patched files.

Well, like it did on the CI, right ?
Or maybe should I add this to the composer command ?

Reply:
I have an idea, I will move the code in another function and mimic the behavior of setUp()

Reply:
Only UboDeclarationsTest is really affected by the patch, DisputesTest is commented out code
I am not sure if you think that it's okay to add a call in each function to mimic setup, will try that anyway